src/pokeradar.c

```diff
@@ -0,0 +1,590 @@
+{
+    UnkStruct_020698E4 * v0;
+    int v1;
+    int v2;
+
+    v0 = sub_0202D830(sub_0202D834(param0->unk_0C));
+    v1 = v0->unk_00[chain->unk_D0].unk_02;
+
+    if (v1 < chain->count) {
+        v0->unk_00[chain->unk_D0].unk_02 = chain->count;
+        v0->unk_00[chain->unk_D0].unk_00 = chain->species;
+
+        ov6_02243160(v0);
+
+        if (v0->unk_00[chain->unk_D0].unk_02 <= chain->count) {
+            for (v2 = 0; v2 < 3; v2++) {
+                if (v0->unk_00[(2 - v2)].unk_02 == chain->count) {
+                    chain->unk_D0 = (2 - v2);
+                    return;
+                }
+            }
+
+            GF_ASSERT(0);
+        }
+    }
+}
+
+static u8 sub_0206994C (UnkStruct_0203CDB0 * param0)
+{
+    UnkStruct_020698E4 * v0;
+    u8 v1;
+    BOOL v2;
+
+    v0 = sub_0202D830(sub_0202D834(param0->unk_0C));
+
+    for (v1 = 0; v1 < 3; v1++) {
+        if (v0->unk_00[v1].unk_00 == 0) {
+            return v1;
+        }
+    }
+
+    (v0->unk_00[0].unk_02 < v0->unk_00[1].unk_02) ? (v2 = 1) : (v2 = 0);
+
+    if (v2) {
+        v1 = 0;
+    } else {
+        v1 = 1;
+    }
+
+    (v0->unk_00[v1].unk_02 < v0->unk_00[2].unk_02) ? (v2 = 1) : (v2 = 0);
+
+    if (!v2) {
+        v1 = 2;
+    }
+
+    return v1;
+}
+
+static BOOL CheckPatchContinueChain (const u8 patchRing, const int battleResult) {
+    u8 * rates;
+    u8 ratesNormal[4] = {88, 68, 48, 28};
+    u8 ratesBoosted[4] = {98, 78, 58, 38};
+
+    if (battleResult == 0x1) {  // If the battle resulted in fainting the mon, use the regular rates
+        rates = ratesNormal;
+    } else if (battleResult == 0x4) {  // If the battle resulted in a capture, use the boosted rates
+        rates = ratesBoosted;
+    }
+
+    if (inline_020564D0(100) < rates[patchRing]) {  // Check if random number falls within the rates
+        return TRUE;  // Patch will continue the chain
+    } else {
+        return FALSE;  // Patch will break the chain
+    }
+}
+
+BOOL RefreshRadarChain (UnkStruct_020508D4 * param0) {
+    UnkStruct_0203CDB0 * v0 = sub_02050A60(param0);
+    int * v1 = sub_02050A64(param0);
+
+    switch (*v1) {
+    case 0:
+    {
+        u8 * v2;
+
+        sub_02062C48(v0->unk_38);
+
+        v2 = sub_0202D9C4(sub_0202D834(v0->unk_0C));
+
+        if ((*v2) < RADAR_BATTERY_STEPS) {
+            sub_0203E8E0(param0, 8970, NULL, NULL);
+            *(u16 *)(sub_0203F098(v0, 41)) = RADAR_BATTERY_STEPS - (*v2);
+            (*v1) = 4;
+        } else {
+            int v3, v4;
+
+            (*v2) = 0;
+
+            v3 = sub_0205EABC(v0->unk_3C);
+            v4 = sub_0205EAC8(v0->unk_3C);
+
+            RadarSpawnPatches(v0, v3, v4, v0->chain);
+
+            if (v0->chain->active) {
+                SetupGrassPatches(v0, 0x1, v0->chain);
+
+                sub_02069638(v0, v0->chain);
+                (*v1) = 1;
+            } else {
+                (*v1) = 3;
+            }
+        }
+    }
+    break;
+    case 1:
+        Sound_PlayBGM(1150);
```

Comment:
Can you create an `enum` definition for this in `constants/sdat.h`?

Reply:
Remind me some time, and I'll fill out the other songs in sdat.h